src/cpp/debug.cpp

```diff
@@ -15,129 +16,115 @@ extern Vector *accelerations;
 extern Vector *velocities;
 extern Vector *displacements;
 
-// Output file streams.
-std::ofstream particles_file;
-std::ofstream properties_file;
-std::ofstream contacts_file;
-std::ofstream normal_forces_file;
-std::ofstream tangent_forces_file;
-std::ofstream forces_file;
-std::ofstream accelerations_file;
-std::ofstream velocities_file;
-std::ofstream displacements_file;
+void write_debug_information(const int step, const size_t num_particles,
+                             const size_t contacts_size, const char *debug_folder) {
 
-void write_debug_information(const int step, const int particle_num,
-                             const size_t num_particles, const size_t contacts_size,
-                             const char *debug_folder) {
-
-  open_files(step, particle_num, debug_folder);
+  std::ofstream debug_file(std::string(debug_folder) + "/"
+                           + "step_" + std::to_string(step),
+                      std::ios::out | std::ios::trunc);
 
-  // Write the contents of each data structure to its corresponding file.
-  // Write the header of each file.
-  write_header(step, particle_num, "Particles.",
-               "x_coordinate y_coordinate raidus", particles_file);
-  write_header(step, particle_num, "Particle Properties.",
-               "mass kn ks", properties_file);
-  write_header(step, particle_num, "Contacts.",
-               "p1_idx p2_idx overlap", contacts_file);
-  write_header(step, particle_num, "Normal forces.",
-               "", normal_forces_file);
-  write_header(step, particle_num, "Tangent forces.",
-               "", tangent_forces_file);
-  write_header(step, particle_num, "Forces.",
-               "x_component y_component", forces_file);
-  write_header(step, particle_num, "Accelerations.",
-               "x_component y_component", accelerations_file);
-  write_header(step, particle_num, "Velocities.",
-               "x_component y_component", velocities_file);
-  write_header(step, particle_num, "Displacements.",
-               "x_component y_component", displacements_file);
+  int hola = contacts_size;
```

Comment:
Please rename this variable.

More than that. why it is even there? It is not used.

Reply:
Sorry, I was fixing a compilation error due that _contacts\_size_ was not used